lib/src/parse/stylesheet.dart

```diff
@@ -1135,54 +1135,100 @@ abstract class StylesheetParser extends Parser {
     return url.startsWith("http://") || url.startsWith("https://");
   }
 
-  /// Consumes a supports condition and/or a media query after an `@import`.
+  /// Consumes a sequence of modifiers (such as media or supports queries)
+  /// after an import argument.
   ///
-  /// Returns `null` if neither type of query can be found.
-  Tuple2<SupportsCondition?, Interpolation?>? tryImportQueries() {
-    SupportsCondition? supports;
-    if (scanIdentifier("supports")) {
-      scanner.expectChar($lparen);
-      var start = scanner.state;
-      if (scanIdentifier("not")) {
-        whitespace();
-        supports = SupportsNegation(
-            _supportsConditionInParens(), scanner.spanFrom(start));
-      } else if (scanner.peekChar() == $lparen) {
-        supports = _supportsCondition();
-      } else {
-        if (_lookingAtInterpolatedIdentifier()) {
-          var identifier = interpolatedIdentifier();
-          if (identifier.asPlain?.toLowerCase() == "not") {
-            error('"not" is not a valid identifier here.', identifier.span);
-          }
+  /// Returns `null` if there are no modifiers.
+  Interpolation? tryImportModifiers() {
+    // Exit before allocating anything if we're not looking at any modifiers, as
+    // is the most common case.
+    if (!_lookingAtInterpolatedIdentifier() && scanner.peekChar() != $lparen) {
+      return null;
+    }
 
-          if (scanner.scanChar($lparen)) {
-            var arguments = _interpolatedDeclarationValue(
-                allowEmpty: true, allowSemicolon: true);
-            scanner.expectChar($rparen);
-            supports = SupportsFunction(
-                identifier, arguments, scanner.spanFrom(start));
+    var start = scanner.state;
+    var buffer = InterpolationBuffer();
+    while (true) {
+      if (_lookingAtInterpolatedIdentifier()) {
+        if (!buffer.isEmpty) buffer.writeCharCode($space);
+
+        var identifier = interpolatedIdentifier();
+        buffer.addInterpolation(identifier);
+
+        var name = identifier.asPlain?.toLowerCase();
+        if (name != "and" && scanner.scanChar($lparen)) {
+          if (name == "supports") {
+            var query = _importSupportsQuery();
+            if (query is! SupportsDeclaration) buffer.writeCharCode($lparen);
+            buffer.addInterpolation(query.toInterpolation());
+            if (query is! SupportsDeclaration) buffer.writeCharCode($rparen);
           } else {
-            // Backtrack to parse a variable declaration
-            scanner.state = start;
+            buffer.writeCharCode($lparen);
+            buffer.addInterpolation(_interpolatedDeclarationValue(
+                allowEmpty: true, allowSemicolon: true));
+            buffer.writeCharCode($rparen);
+          }
+
+          scanner.expectChar($rparen);
+          whitespace();
+        } else {
+          whitespace();
+          if (scanner.scanChar($comma)) {
```

Comment:
This support for comma looks incompatible with using the comma as separator between multiple ImportArgument.

Reply:
This actually exists to match the previous behavior: because `MediaQueryList` can contain commas at nearly any point, you could already write `@import "..." foo, bar, baz`. This just makes it more explicit.